### PR TITLE
Fix Inductor Stride Mismatch for torch.fft.ifft 

### LINKED ISSUE #166514: Inductor Fails with Stride Mismatch for `torch.fft.ifft` Due to Incorrect Meta Kernel Implementation
### 🐛 Describe the bug

When compiling a model containing `torch.fft.ifft` operations with `torch.compile(backend="inductor")`, a stride validation error occurs due to an incorrect fake (meta) kernel implementation. The identical model works correctly with `eager` and `aot_eager` backends.

### To reproduce
```python
import torch
import torch.nn as nn

class FFTModel(nn.Module):
    def __init__(self):
        super().__init__()
        self.conv = nn.Conv2d(3, 64, kernel_size=3, padding=1)
    
    def forward(self, x):
        x = self.conv(x)
        x = torch.fft.ifft(x, n=256)
        x = torch.real(x)
        return x

model = FFTModel()
x = torch.randn(4, 3, 32, 32)

backends = ["eager", "aot_eager", "inductor"]

for backend in backends:
    try:
        compiled_model = torch.compile(model, backend=backend)
        output = compiled_model(x)
        print(f"✅ '{backend}': Success, output shape {output.shape}")
    except Exception as e:
        print(f"❌ '{backend}': Failed - {str(e)}")
```

### Error logs

```
✅ 'eager': Success, output shape torch.Size([4, 64, 32, 256])
✅ 'aot_eager': Success, output shape torch.Size([4, 64, 32, 256])
❌ 'inductor': Failed - expected size 64==64, stride 256==8192 at dim=1; expected size 32==32, stride 16384==256 at dim=2
Error in op: torch.ops.aten._fft_r2c.default
This error most often comes from a incorrect fake (aka meta) kernel for a custom op.
Use torch.library.opcheck to test your custom op.
See https://pytorch.org/docs/stable/library.html#torch.library.opcheck
```

### Versions

Collecting environment information...
PyTorch version: 2.10.0.dev20251029+cpu
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 11
GCC version: Could not collect
Clang version: Could not collect
CMake version: version 4.0.2
Libc version: N/A

Python version: 3.10.10 (tags/v3.10.10:aad5f6a, Feb  7 2023, 17:20:36) [MSC v.1929 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-10-10.0.26100-SP0
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @aakhundov @coconutruben

### PR DESCRIPTION
**Pull request by ahkush**
Fixes #166514

**Problem**

When FFT operations follow convolution layers in a compiled model, the inductor backend may optimize conv2d outputs to non-contiguous layouts. FFT operations decompose into ```constant_pad_nd``` for input resizing, which then receives non-contiguous tensors, causing stride validation errors like:
```expected size 64==64, stride 256==8192 at dim=1```
```Error in op: torch.ops.aten._fft_r2c.default```

**Solution**

Added ```ir.ExternKernel.require_contiguous(x)``` at the beginning of the ```constant_pad_nd``` lowering function. This ensures padding operations receive contiguous inputs, preventing stride mismatches when tensors flow through FFT decompositions. 
Test case added to verify correct behavior with conv2d → FFT patterns.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo @chenyang78

@vishalgoyal316 @cleonard530 @abhitorch81 @gaurav-redhat @morrison-turnansky @skpark-rh

### CODE REVIEW

File: test/inductor/test_torchinductor.py [hunk 1]
```diff
@@ -12025,6 +12025,31 @@ def fn(x):
...
 
+    def test_fft_ifft_with_conv(self):
+        class FFTModel(nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.conv = nn.Conv2d(3, 64, kernel_size=3, padding=1)
+            
+            def forward(self, x):
+                x = self.conv(x) 
+                x = torch.fft.ifft(x, n=256)
```
Comment (zou3519) [inline test/inductor/test_torchinductor.py:12036]: I thought the problem was that a meta kernel or decomposition was wrong somewhere, not that we needed to force Inductor to make the strides contiguous?

Comment (morrison-turnansky) [reply test/inductor/test_torchinductor.py:12036]: I was suspicious as well when I saw the fix. That is why I wanted to see if the bug still existed on viable/strict and verified repro was no longer valid.

**Review by eellison — CHANGES_REQUESTED** [review]
please update the meta implementation instead, which is the root cause of the issue
